src/components/Common/StationData.jsx

```diff
@@ -128,10 +127,11 @@ const StationData = () => {
     stationTypeFilter,
     tog,
   ]);
-    
-  function handleSelectStation ({position, _id}) {
-    setSelectedStation(position)
-    setSelectedStationId(_id)
+
+  function handleSelectStation({ position, _id }) {
+    setSelectedStation(position);
+    setSelectedStationId(_id);
+    onStationClick(_id); // to call the onStationClick prop
```

Comment:
onStationClick should be attached to the station name div, not the station container div. It shouldnt be with these other 2 set states

Reply:
Thanks, have moved to the h2 element. Let me know if further action required.